### PR TITLE
chore: bump go-corset to latest version and tidy dependencies

### PR DESCRIPTION
**Pull request by gusiri**
This PR bump the go-corset version to latest and tidy related dependencies. 

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

File: prover/zkevm/prover/hash/packing/packing_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func makeTestCasePackingModule(uc generic.HashingUsecase) (
 	var (
 		// max number of blocks that can be extracted from limbs
 		// if the number of blocks passes the max, newPack() would panic.
-		maxNumBlock = 35
+		maxNumBlock = 36
```
Comment (gusiri) [inline prover/zkevm/prover/hash/packing/packing_test.go:21]: --- FAIL: TestPackingModule (0.13s)
    --- FAIL: TestPackingModule/MiMC (0.02s)
panic: the number of the blocks 36 passes the limit 35 [recovered]
	panic: the number of the blocks 36 passes the limit 35
	
@AlexandreBelling Do you have any idea what caused this change?

Comment (gusiri) [reply prover/zkevm/prover/hash/packing/packing_test.go:21]: I guess previously the `number of blocks` was counted as 35, but now it's counted as 36, which caused it to exceed the `maxNumBlock` of 35.

Comment (AlexandreBelling) [reply prover/zkevm/prover/hash/packing/packing_test.go:21]: Oh right I did a small change regarding this. I did not see it broke the tests

**Review by AlexandreBelling — APPROVED** [review]

File: prover/zkevm/prover/hash/packing/packing_test.go (hunk 1 above)
Comment (AlexandreBelling) [reply prover/zkevm/prover/hash/packing/packing_test.go:21]: I think it might have been skipped by the CI previously and this PRs makes it happen again

**Review by arijitdutta67 — APPROVED** [review]

**Pull request merged**